src/common/records.lua

```diff
@@ -7,27 +7,41 @@ Records = Records
 		["@"] = {
 			transactionId = "-k7t8xMoB8hW482609Z9F4bTFMC3MnuW8bTvTyT8pFI",
 			ttlSeconds = 900,
+			priority = nil,
 		},
 	}
 
 --- Set a record in the Records of the ANT.
 ---@param name string The name of the record.
 ---@param transactionId string The transaction ID of the record.
 ---@param ttlSeconds number The time-to-live in seconds for the record.
+---@param priority integer|nil The sort order of the record
 ---@return Record
-function records.setRecord(name, transactionId, ttlSeconds)
+function records.setRecord(name, transactionId, ttlSeconds, priority)
 	utils.validateUndername(name)
 	assert(utils.isValidArweaveAddress(transactionId), "Invalid Arweave ID")
 	utils.validateTTLSeconds(ttlSeconds)
+	assert(
+		priority == nil or math.type(priority) == "integer",
+		"Priority must be an integer or nil, received " .. tostring(priority)
+	)
+	-- TODO: verify this is a desired behaviour
+	if name == "@" then
+		assert(priority == nil, "Cannot set priority to @ record")
+	end
 
+	collectgarbage("stop")
```

Comment:
Seems like GC is performing an operation *while* we modify this variable, unclear to what the core cause is after a thorough investigation

There may be some EMSCRIPTEN memory allocation things going on here that is triggering this - bumping the stack size can fix this problem, but considering its not necessary, seems to indicate rather than a deficiency, its a mismanagement.

Reply:
i'm confused why this operation causes the issue. do we need to allocate more to the stack in general to avoid constantly bumping into this problem?

Reply:
Yeah I think we do. Its just as of right now we need to remain backwards compatible with this limit of the stack - once we can reassign to evolve we can start messing with the stack memory. Until then we are restricted by what we have deployed previously.